packages/types/src/codec/Compact.spec.ts

```diff
@@ -109,52 +109,52 @@ describe('Compact', (): void => {
   describe('constructor', (): void => {
     it('has the correct bitLength for constructor values (BlockNumber)', (): void => {
       expect(
-        new Compact(ClassOf('BlockNumber'), 0xfffffff9).bitLength()
+        new (Compact.with(ClassOf(registry, 'BlockNumber')))(registry, 0xfffffff9).bitLength()
       ).toEqual(32);
     });
 
     it('has the correct encodedLength for constructor values (string BlockNumber)', (): void => {
       expect(
-        new Compact('BlockNumber', 0xfffffff9).encodedLength
+        new (Compact.with('BlockNumber'))(registry, 0xfffffff9).encodedLength
       ).toEqual(5);
     });
 
     it('has the correct encodedLength for constructor values (class BlockNumber)', (): void => {
       expect(
-        new Compact(ClassOf('BlockNumber'), 0xfffffff9).encodedLength
+        new (Compact.with(ClassOf(registry, 'BlockNumber')))(registry, 0xfffffff9).encodedLength
```

Comment:
Looking at these, we could put Compact, Uint, Vec, Tuple, Struct etc as abstract classes, and enforce the usage `.with()` which will create a real class

Reply:
Throwing ideas again, an alternative syntax would be `new Compact.with(registry.BlockNumber)`, on in general `new (api.)registry.BlockNumber('234')`

Reply:
I would prefer them to be used that way, yes. Probably after https://github.com/polkadot-js/api/pull/1518 (which has it's own issues, but opens the way for removing the `U{8,16,32,64,128,256}` classes (and allows stuff like `U2048`)